src/sdk/pynni/nni/compression/torch/pruning/amc/amc_pruner.py

```diff
@@ -0,0 +1,325 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import os
+from copy import deepcopy
+from argparse import Namespace
+import numpy as np
+import torch
+from tensorboardX import SummaryWriter
```

Comment:
could you check whether i have to install `tensorboardX` even i only use other pruners?

Reply:
Only amc is using `tensorboardX `

Reply:
i know, i am wondering whether i have to install `tensorboardX` if i only use other pruners